test/test_core.py

```diff
@@ -7454,7 +7454,7 @@ def test_demangle_stacks_symbol_map(self):
     self.assertIsNotNone(full_aborter)
     self.assertIsNotNone(short_aborter)
     print('full:', full_aborter, 'short:', short_aborter)
-    if config.SPIDERMONKEY_ENGINE and os.path.exists(config.SPIDERMONKEY_ENGINE[0]):
+    if self.js_engines[0] == config.SPIDERMONKEY_ENGINE:
```

Comment:
```suggestion
    if config.SPIDERMONKEY_ENGINE in self.js_engines:
```
Sorry if I wasn't clear before, this is what I meant. This should handle `js_engines = [NODE_JS, SPIDERMONKEY_ENGINE, ..]` etc. - which I think we want to support?

Reply:
Yes that would make sense i think.. trying it out though it looks like this doesn't currently pass with current spider monkey.  I will see if I can make it pass and get this test running under spidermonkey in CI.